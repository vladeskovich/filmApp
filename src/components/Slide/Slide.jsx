import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';
import Typography from '../Typography';
import Genres from '../Genres';
import getTimeFromMin from '../../utils/getTimeFromMin';
import Rating from '../Rating';
import styles from './Slide.scss';
import imagePath from '../../constants/imagePath';

const Slide = ({
  slideImage,
  titleFilm,
  voteCount,
  runtime,
  genres,
}) => {
  const time = getTimeFromMin(runtime);
  const urlImage = imagePath + slideImage;

  return (
    <div style={{ backgroundImage: `url(${urlImage})` }} className={styles.slide}>
      <div className={styles.slideTop}></div>
      <div className={styles.slideBottom}>
        <div className={styles.slideInfo}>
          <div className={styles.slideTitle}>
            <Typography className={styles.textTitle}>{titleFilm.toUpperCase()}</Typography>
          </div>
          <Genres data={genres}/>
          <div className={styles.wrapperRuntimeFilm}>
            <Typography className={styles.runtimeFilm}>
              | {time.hours}h {time.minutes}m
            </Typography>
          </div>
          <Rating rating={voteCount}/>
        </div>
        <div className={styles.slideButtons}>
          <Button
            color='blue'
            outline
          >Watch Now</Button>
          <Button
            color='none'
            outline
          >View Info</Button>
        </div>
      </div>
    </div>
  );
};
Slide.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string,
  slideImage: PropTypes.string,
  titleFilm: PropTypes.string,
  voteCount: PropTypes.number,
  runtime: PropTypes.number,
  genres: PropTypes.array,
};

Slide.defaultProps = {
  slideImage: ' ',
  titleFilm: ' ',
  voteCount: 0,
  runtime: 0,
  genres: [],
};

export default Slide;
