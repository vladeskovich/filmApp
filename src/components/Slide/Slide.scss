.slide {
  background-color: grey;
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background-size: cover;
  color: white;
  box-shadow: inset 150px 150px 150px rgba(0, 0, 0, 0.5), inset -150px -150px 150px rgba(0, 0, 0, 0.5);
}

.slideBottom {
  display: flex;
  justify-content: space-between;
  padding-bottom: 20px;
}

.slideInfo {
  padding-left: 70px;
  width: 100%;
  max-width: 500px;
}

.textTitle {
  font-size: 45px;
  font-weight: bold;
  color: white;
}

.wrapperRuntimeFilm {
  display: inline-block;
}

.runtimeFilm {
  font-size: 14px;
  font-weight: normal;
}

.slideButtons {
  position: relative;
  display: flex;
  width: 300px;
  padding-right: 100px;
  justify-content: space-between;
  align-items: flex-end;
}

.filmOverview {
  position: absolute;
  bottom: 50px;
  left: 0;
  overflow: auto;
  box-sizing: border-box;
  width: 300px;
  max-height: 180px;
  padding: 13px;
  transition: opacity 0.45s ease;
  opacity: 0;
  line-height: 2;
  background: rgba(0, 0, 0, 0.75);

  .textOverview {
    font-size: 16px;
  }

  &::-webkit-scrollbar {
    display: none;
  }
}

.filmOverview.show {
  opacity: 1;
}

@media screen and (max-width: 768px) {
  .slideBottom {
    max-width: 90%;
    flex-direction: column;
    padding-left: 10px;
  }

  .slideInfo {
    padding-left: 0;
  }

  .textTitle {
    font-size: 30px;
  }

  .slideButtons {
    width: 100%;
    padding-right: 0;
    padding-top: 15px;
    justify-content: space-around;
  }
}
