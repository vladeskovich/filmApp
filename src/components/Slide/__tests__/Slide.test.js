import React from 'react';
import Slide from '../Slide';

describe('Slide', () => {
  const props = {
    slideImage: '/w56545447547745',
    titleFilm: 'Inferno',
    voteCount: 8,
    runtime: 145,
    genres: [
      { name: 'Action' },
      { name: 'Adventure' },
    ],
    onShow: () => {},
    id: 1,
  };

  it('should render Slide component with props', () => {
    const component = shallow(<Slide {...props}/>);

    expect(component).toMatchSnapshot();
  });
});
