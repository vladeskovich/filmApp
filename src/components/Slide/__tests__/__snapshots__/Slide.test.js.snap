// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide should render Slide component with props 1`] = `
<div
  className="Slide__slide_1PvkpWrD"
  style={
    Object {
      "backgroundImage": "url(http://image.tmdb.org/t/p/w1280/w56545447547745)",
    }
  }
>
  <div />
  <div
    className="Slide__slideBottom_2VWBG3Wm"
  >
    <div
      className="Slide__slideInfo_3a__zWUv"
    >
      <div>
        <Typography
          className="Slide__textTitle_14gSXKlP"
          variant="body"
        >
          INFERNO
        </Typography>
      </div>
      <Genres
        data={
          Array [
            Object {
              "name": "Action",
            },
            Object {
              "name": "Adventure",
            },
          ]
        }
      />
      <div
        className="Slide__wrapperRuntimeFilm_36X8-1RM"
      >
        <Typography
          className="Slide__runtimeFilm_2NI41r4E"
          variant="body"
        >
          | 
          2
          h 
          25
          m
        </Typography>
      </div>
      <Rating
        rating={8}
      />
    </div>
    <div
      className="Slide__slideButtons_29LU2FWG"
    >
      <Button
        color="blue"
        data-item-id={1}
        onClick={[Function]}
        outline={true}
        type="button"
      >
        Watch Now
      </Button>
      <Button
        color="none"
        outline={true}
        type="button"
      >
        View Info
      </Button>
    </div>
  </div>
</div>
`;
